core/src/main/java/feign/DefaultMethodHandler.java

```diff
@@ -50,12 +51,49 @@ public DefaultMethodHandler(Method defaultMethod) {
   private Lookup readLookup(Class<?> declaringClass)
       throws IllegalAccessException, InvocationTargetException, NoSuchFieldException {
     try {
-      return safeReadLookup(declaringClass);
-    } catch (NoSuchMethodException e) {
+      if (System.getProperty("java.vm.name").equalsIgnoreCase("Dalvik")) {
+        return androidLookup(declaringClass);
```

Comment:
this entire flow is not been tested at all, as we only build feign with openjdk.

Is that something we could include on circleCI?

Reply:
Yes CircleCI allows to run under Dalvik/ART.

Build of my project was OK in CCI with the changes to feign-core

I'll look into if/how the flow can tested against Android's JVM's, for this module or perhaps for the whole project if it works out more easy to just add a line to the .yml specifying Dalvik/ART.

Reply:
Hmm my most recent changes seem to have gone missing in the android-compat branch. Will update it later.